src/main/java/io/github/thebusybiscuit/slimefun4/utils/SlimefunUtils.java

```diff
@@ -321,6 +332,19 @@ public static boolean isItemSimilar(@Nullable ItemStack item, @Nullable ItemStac
         }
     }
 
+    private static @Nullable String getMetaId(ItemMeta itemMeta) {
+        Optional<String> optionalId = Slimefun.getItemDataService().getItemData(itemMeta);
+        return optionalId.orElse(null);
+    }
+
+    private static @Nonnull Optional<DistinctiveItem> getDistinctiveItem(@Nonnull String id) {
```

Comment:
Not sure this method is needed. Feels like it should just be moved inside isItemSimilar

Reply:
isItemSimilar is already so big I really didn't want to add this directly inside of it - since this comment I am now using this an additional 2 times also to I think this may be best kept outside, just to keep this monster method readable :D